core/shared_queue.js

```diff
@@ -180,7 +180,6 @@ SharedQueue Binary Layout
   }
 
   function dispatch(opId, control, zeroCopy = null) {
-    maybeInit();
```

Comment:
This is very tricky and brittle, ideally we'd want to remove this completely, but if tests are green for now I'm okay 👍

Reply:
We shove `maybeInit()` eveywhere because we weren't sure... clearly it is overkill.  I can't see of a code path that gets here and initting the shared queue during snapshotting causes a serialization problem.

Reply:
That's right it's very convoluted. I already started playing with completely removing `shared_queue` but surprisingly I get horrible performance and need to put some more work. Ditto, if it passes tests I'm okay with that